apps/wing-console/packages/use-persistent-state/src/index.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useRef,
+  MutableRefObject,
+  useState,
+} from "react";
+
+const PersistentStateContext = createContext<
+  | {
+      state: MutableRefObject<Map<string, any[]>>;
+    }
+  | undefined
+>(undefined);
+
+export const usePersistentStateContext = () => {
+  const context = useContext(PersistentStateContext);
+  if (!context?.state) {
+    throw new Error(
+      "usePersistentState must be used within a PersistentStateProvider",
+    );
+  }
+  return context.state;
+};
+
+export const PersistentStateProvider = (props: PropsWithChildren) => {
+  const state = useRef(new Map<string, any[]>());
+  return (
+    <PersistentStateContext.Provider value={{ state }}>
+      {props.children}
+    </PersistentStateContext.Provider>
+  );
+};
+
+export const createPersistentState = (prefix: string) => {
```

Comment:
the prefix is actually the unique id for the state, right? why call it a prefix?

Reply:
true! it make no sense now.

fixed

Reply:
Initially it make sense but yeah, we should name it key or id.